### PR TITLE
Feature xcmp assets

### PR DESCRIPTION
**Pull request by akru**
**Pull Request Summary**

This PR introduces XC20 precompiles adopted to be used in Astar ecosystem.

**Check list**
- [ ] contains breaking changes
- [ ] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tests and/or benchmarks are included
- [ ] changed API client type definition or chain metadata

**This pull request makes the following changes:**

**Adds**
- (ex: Add feature A)

**Fixes**
- (ex: Fix validation function)

**Changes**
- (ex: Change document B)

**To-dos**
> *Feel free to remove this section if it's not applicable

- [ ] (ex: add user list)

### CODE REVIEW

**Review by hoonsubin — APPROVED** [review]
So this PR is about importing MB ERC20 precompiled? I think we should change the PR title to reflect that.

LGTM!

(please resolve the unit test issue)

File: precompiles/assets-erc20/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+    fn account_to_asset_id(account: Account) -> Option<AssetId>;
+
+    // Get AccountId from AssetId
+    fn asset_id_to_account(asset_id: AssetId) -> Account;
+}
+
+/// The following distribution has been decided for the precompiles
+/// 0-1023: Ethereum Mainnet Precompiles
+/// 1024-2047 Precompiles that are not in Ethereum Mainnet but are neither Moonbeam specific
+/// 2048-4095 Moonbeam specific precompiles
```
Comment (Dinonard) [inline precompiles/assets-erc20/src/lib.rs]: Please change this.

**Review by Dinonard — CHANGES_REQUESTED** [review]
Please fix the UT.

File: precompiles/assets-erc20/src/lib.rs (hunk 1 above)
Comment (akru) [reply precompiles/assets-erc20/src/lib.rs]: Changed

**Comment by akru** [comment]
@Dinonard please review, most comments fixed.

File: precompiles/assets-erc20/src/lib.rs (hunk 1 above)
Comment (Dinonard) [reply precompiles/assets-erc20/src/lib.rs]: Great!

File: precompiles/utils/macro/tests/tests.rs [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
 // the Free Software Foundation, either version 3 of the License, or
 // (at your option) any later version.
 
-// Moonbeam is distributed in the hope that it will be useful,
+// Utils is distributed in the hope that it will be useful,
 // but WITHOUT ANY WARRANTY; without even the implied warranty of
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
 
-use sha3::{Digest, Keccak256};
```
Comment (Dinonard) [inline precompiles/utils/macro/tests/tests.rs:14]: CI UT is failing because of the `Keccak256` removal.

Comment (akru) [reply precompiles/utils/macro/tests/tests.rs:14]: Fixed

**Review by Dinonard — APPROVED** [review]
LGTM

**Pull request merged**